narwhal/types/src/primary.rs

```diff
@@ -61,24 +61,24 @@ fn now() -> TimestampMs {
     }
 }
 
-// Additional metadata information for an entity. Those data
+// Additional metadata information for an entity.
+//
+// The structure as a whole is not signed. As a result this data
 // should not be treated as trustworthy data and should be used
 // for NON CRITICAL purposes only. For example should not be used
 // for any processes that are part of our protocol that can affect
 // safety or liveness.
-#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Arbitrary, MallocSizeOf)]
+//
+// However selected fields, such as
+// - Header::metadata::created_at
+// are signed to support commit timestamps.
+#[derive(Clone, Default, Serialize, Deserialize, Debug, PartialEq, Eq, Arbitrary, MallocSizeOf)]
```

Comment:
Initially intention for this Metadata struct was to put fields that are not covered by signatures.

I feel like right now this Metadata struct might be too confusing in terms of what is signed and what is not.

Should we just take following approach:
- Remove the struct
- Add field `created_at` to Header
- Add field `created_at_untrusted` to Batch? (or, alternatively just include created_at into Batch digest as well even though it is not super needed)

Reply:
Even the Header timestamp is untrusted in the current design, the only think we can trust is the median of parent timestamps right?

Reply:
Yeah, but original intention for Metadata struct was not whether it is trusted or not, but the fact whether it is covered by hash/signature or not.